qa-ubuntu

```diff
@@ -0,0 +1,149 @@
+    if microk8s.kubectl get secret marketo &> /dev/null; then microk8s.kubectl delete secret marketo; fi
+    microk8s.kubectl create secret generic marketo --from-literal=api_client='notsosecret' --from-literal=api_secret='notsosecret'
+    
+    # Fake badgr key
+    if microk8s.kubectl get secret badgr &> /dev/null; then microk8s.kubectl delete secret badgr; fi
+    microk8s.kubectl create secret generic badgr --from-literal=user='notsosecret' --from-literal=qa-password='notsosecret' --from-literal=password='notsosecret'
+
+    # Fake cube-edx key
+    if microk8s.kubectl get secret cube-edx &> /dev/null; then microk8s.kubectl delete secret cube-edx; fi
+    microk8s.kubectl create secret generic cube-edx --from-literal=client-secret='notsosecret' --from-literal=qa-client-secret='notsosecret' --from-literal=client-id='notsosecret' --from-literal=qa-client-id='notsosecret'
+}
+
+function add_docker_credentials_microk8s() {
+    if ! microk8s.kubectl get secret registry-access &> /dev/null; then
+
+        username=""
+        password=""
+        while [[ -z "${username}" || -z "${password}" ]]; do
+            echo "##############"
+            echo "Docker registry credentials"
+            echo
+            echo -n "Username: "
+            read username
+            echo -n "Password: "
+            read -s password
+            echo -e "\n##############"
+        done
+
+        microk8s.kubectl create secret docker-registry registry-access --docker-server=prod-comms.docker-registry.canonical.com --docker-username="${username}" --docker-password="${password}" --docker-email=root@localhost
+
+        microk8s.kubectl patch serviceaccount default -p '{"imagePullSecrets": [{"name": "registry-access"}]}' --namespace default
+    fi
+}
+
+function apply_configuration() {
+    if ! microk8s.kubectl get configmap ubuntu-com &> /dev/null; then
+        microk8s.kubectl create configmap ubuntu-com --from-literal=maintenance=local;
+    fi
+}
+
+function enable_ingress() {
+    microk8s.enable ingress dns
+}
+
+function deploy_locally() {
+    environment="${1}"
+    values_file="${2}"
+
+    echo "Deploying project ${values_file}"
+    konf --local-qa ${environment} ${values_file} | microk8s.kubectl apply --filename -
+}
+
+function run() {
+    # Arguments
+    environment="$1"
+    values_file="$2"
+
+    # Mandatory arguments
+    if [ -z "${environment:-}" ]; then invalid "No environment specified (e.g. 'production')"; fi
+    if [ -z "${values_file:-}" ]; then invalid "No values file specified (e.g. 'konf/site.yaml')"; fi
+
+    # Validate arguments
+    if [[ ! "$environment" =~ ^(demo|production|staging)$ ]]; then
+        invalid "You need to specify a valid environment: 'demo', 'production' or 'staging'"
+    fi
+
+    if [[ ! -f $values_file ]] ; then
+        invalid "File '${values_file}' doesn't exits, aborting."
+    fi
+
+    echo "Make sure microk8s is running"
+    echo "Make sure the database container is running"
```

Comment:
These messages are a bit confusing to be echoing out every time. I interpreted them as complaining that these things weren't running, until I realised that it seemed to be carring on fine...